raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -23,13 +23,37 @@
 )
 from raiden.transfer.state import (
     CHANNEL_STATE_OPENED,
-    message_identifier_from_prng,
     RouteState,
     NettingChannelState,
+    message_identifier_from_prng,
 )
-from raiden.constants import MAXIMUM_PENDING_TRANSFERS
+from raiden.transfer.state_change import Block
+from raiden.settings import DEFAULT_NUMBER_OF_CONFIRMATIONS_BLOCK
+from raiden.utils import typing
 
-ChannelMap = typing.Dict[typing.ChannelID, NettingChannelState]
+
+def handle_block(
+        initiator_state: InitiatorTransferState,
+        state_change: Block,
+        channel_state: NettingChannelState,
+        pseudo_random_generator: random.Random,
+) -> TransitionResult:
+    secrethash = initiator_state.transfer.lock.secrethash
+    locked_lock = channel_state.our_state.secrethashes_to_lockedlocks[secrethash]
+    if state_change.block_number < locked_lock.expiration + DEFAULT_NUMBER_OF_CONFIRMATIONS_BLOCK:
+        # Lock still valid
+        return TransitionResult(initiator_state, list())
+
+    # Lock has expired, cleanup...
+    channel.delete_secrethash_endstate(channel_state.our_state, secrethash)
```

Comment:
Because this is changing the channel state machine, I would move this inside the channel state machine, in the `events_for_expired_lock` function.